### PR TITLE
[GraphQL] Make pre-heated cache work for upgraders

### PR DESCRIPTION
**Pull request by alloy**
Part of https://artsyproduct.atlassian.net/browse/LD-286

#native_no_changes

### CODE REVIEW

File: Pod/Classes/GraphQL/ARGraphQLQueryCache.m [hunk 1]
```diff
@@ -34,63 +34,68 @@ + (void)load;
...
+            NSParameterAssert(queryID);
+            NSDictionary *variables = queryParams[@"variables"];
+            NSParameterAssert(variables);
+            NSString *cacheKey = CacheKey(queryID, variables);
+
+            // Skip pre-heated cache entries that already exist.
+            if ([existingCacheEntryKeys indexOfObject:cacheKey] != NSNotFound) {
+                DLog(@"[ARGraphQLQueryCache] Existing pre-heated GraphQL cache entry will be ignored: %@", cacheEntryURL.path);
+                continue;
+            }
```
Comment (alloy) [inline Pod/Classes/GraphQL/ARGraphQLQueryCache.m]: This is the actual functional change.

**Comment by alloy** [comment]
I re-organized the code a bit to not nest so deeply, sorry for the noisy diff that has caused.

File: Pod/Classes/GraphQL/ARGraphQLQueryCache.m [hunk 2]
```diff
@@ -34,63 +34,68 @@ + (void)load;
...
 
-    if (firstLaunch) {
-        dispatch_async(dispatch_get_global_queue(QOS_CLASS_UTILITY, 0), ^{
-            NSURL *preHeatedCache = [[NSBundle mainBundle] URLForResource:@"PreHeatedGraphQLCache" withExtension:nil];
-            NSError *error = nil;
-            NSArray<NSURL *> *cacheEntries = [[NSFileManager defaultManager] contentsOfDirectoryAtURL:preHeatedCache
-                                                                           includingPropertiesForKeys:@[(__bridge NSString *)kCFURLNameKey]
-                                                                                              options:NSDirectoryEnumerationSkipsHiddenFiles
-                                                                                                error:&error];
+    dispatch_async(dispatch_get_global_queue(QOS_CLASS_UTILITY, 0), ^{
```
Comment (ashfurrow) [inline Pod/Classes/GraphQL/ARGraphQLQueryCache.m]: `QOS_CLASS_UTILITY` 😚👌

File: Pod/Classes/GraphQL/ARGraphQLQueryCache.m (hunk 1 above)
Comment (ashfurrow) [reply Pod/Classes/GraphQL/ARGraphQLQueryCache.m]: Right, that makes sense. So we'll keep the first pre-heated cache the user gets and ignore new ones for upgrades.

**Pull request merged**